File: LB-Assignments/Assignment_25/program3.c
/*
Input : iRow = 4  iCol = 4
a  b  c  d
1  2  3  4
a  b  c  d
1  2  3  4

*/

#include<stdio.h>

void Pattern(int iRow, int iCol)
{
    int i = 0, j = 0;
    char ch = '\0';

    for(i =1; i <= iRow; i++)
    {
        if(i % 2 == 0)
        {
            for(j = 1; j <= iCol; j++)
            {
                printf("%d\t",j);
            }
            printf("\n");
        }
        else
        {
            for(j = 1, ch = 'a'; j<= iCol; j++,ch++)
            {
                printf("%c\t",ch);
            }
            printf("\n");
        }
        
    }
}

int main()
{
    int iValue1= 0, iValue2 = 0;

    printf("Enter number of Rows : \n");
    scanf("%d",&iValue1);

    printf("Enter number of columns : \n");
    scanf("%d",&iValue2);

    Pattern(iValue1,iValue2);

    return 0;
}

File: LB-Assignments/Assignment_28/program3.java
/*
iRow = 6   iCol = 6
*  *  *  *  *  *
*           *  *
*        *     *
*     *        *
*  *           *
*  *  *  *  *  *

*/
import java.util.*;

class PatternPrint
{
    public int iRow;
    public int iCol;

    public PatternPrint(int a, int b)
    {
        this.iRow = a;
        this.iCol= b;
    }

    void Pattern()
    {
        if(iRow != iCol)
        {
            System.out.println("Invalid Input");
            System.out.println("Number of rows and columns must be same");
            return;
        }

        int i = 0, j = 0;

        for(i = iRow; i >= 1; i--)
        {
            for(j = 1; j<= iCol; j++)
            {
                if((i == j) || (i==1)||(j==1)||(i==iRow)||(j==iCol))
                {
                    System.out.print("*\t");
                }
                else
                {
                    System.out.print("\t");
                }
            }
            System.out.println();
        }
    }
}
class program3
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        int iValue1= 0, iValue2 = 0;

        System.out.println("Enter number of rows :");
        iValue1 = sobj.nextInt();

        System.out.println("Enter number of columns :");
        iValue2 = sobj.nextInt();

        PatternPrint pobj = new PatternPrint(iValue1, iValue2);

        pobj.Pattern();

    }
}